src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -2,18 +2,104 @@
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int INITIAL_CAPACITY = 16;
+    private static final int CAPACITY_MULTIPLIER = 2;
+    private Node<K, V>[] map;
+    private int size;
+    private double threshold;
+
+    public MyHashMap() {
+        this.map = (Node<K, V>[]) new Node[INITIAL_CAPACITY];
+        this.threshold = INITIAL_CAPACITY * LOAD_FACTOR;
+        this.size = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        if (size >= threshold) {
+            resize();
+        }
+        Node<K, V> newNode = new Node<>(key, value);
+        int index = newNode.hash % map.length;
+        if (map[index] == null) {
+            map[index] = newNode;
+            size++;
+        } else {
+            Node<K, V> iterationNode = map[index];
+            while (iterationNode.next != null) {
+                if (isSameKey(iterationNode, key)) {
+                    iterationNode.value = newNode.value;
+                    break;
+                }
+                iterationNode = iterationNode.next;
+            }
+            if (isSameKey(iterationNode, key)) {
+                iterationNode.value = newNode.value;
+            } else {
+                iterationNode.next = newNode;
+                size++;
+            }
```

Comment:
repeatable if clause, u can reassign the 'iterationNode.value' in the first if and put return, then second if-else isn't needed